docs/jwst/error_propagation/main.rst

```diff
@@ -0,0 +1,191 @@
+of all variances whenever a new type of variance array is added to the pipeline
+flow or whenever any of the variance arrays is updated by a step.
+
+
+Stage 1 pipelines 
+-----------------
+Stage 1 pipelines perform detector-level corrections and ramp fitting for
+individual exposures, for nearly all imaging and spectroscopic modes. Details 
+of the pipelines can be found at :ref:`Stage 1 Pipelines <calwebb_detector1>`
+
+Of all of the Stage 1 pipeline steps, below are the particular steps that alter the ERR,
+VAR_POISSON, or VAR_RNOISE arrays of the science countrate data and output those
+altered arrays in the science product that is passed to the next step in the
+pipeline.  For the steps not listed below, those arrays do not exist or are
+propagated unchanged to the next pipeline step.
+
+
+Ramp_fitting
+++++++++++++
+This step calculates and populates the VAR_POISSON and VAR_RNOISE arrays
+(the variances associated with ramp fitting) in the 'rate' and
+'rateints' files, and updates the ERR arrays in those files. The details of the 
+calculations can be found at :ref:`ramp_fitting <ramp_fitting_step>`.
+
+
+Gain_scale
+++++++++++
+The gain_scale step is applied after the ramp_fitting step, and applies to both the 
+rate and rateints products from ramp_fit. The correction is applied to the ERR, 
+VAR_POISSON, and VAR_RNOISE arrays.  The details of the calculations can be
+found at :ref:`gain_scale <gain_scale_step>`.
+
+
+Stage 2 pipelines 
+-----------------
+Stage 2 pipelines perform additional instrument-level and observing-mode corrections and 
+calibrations to produce fully calibrated exposures. There are two Stage 2 pipelines: one 
+for imaging :ref:`calwebb_image2 <calwebb_image2>` and one for 
+spectroscopy :ref:`calwebb_spec2 <calwebb_spec2>`.
+
+In these pipelines, the various steps that apply corrections and calibrations
+apply those same corrections/calibrations to all variance arrays.
+
+
+Stage 2 Imaging Processing pipeline - calwebb_image2, calwebb_tso-image2, calwebb_wfs-image2
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+For the Stage 2 imaging pipelines, below are the steps that alter the ERR or
+VAR arrays of the science countrate data and output those altered arrays in the
+science product that is passed to the next step in the pipeline.  For the steps
+not listed below, those arrays do not exist or are propagated unchanged to the
+next pipeline step.
+
+Flat_field
+~~~~~~~~~~
+The SCI array from the flat-field reference file is divided into the VAR_POISSON
+and VAR_RNOISE arrays of the science data set. The variance in the flat is
+calculated in this step; the ERR array is then recalculated as these three 
+variances added in quadrature.  The newly created VAR_FLAT array and the updated 
+VAR_POISSON and VAR_RNOISE are output to the science products(s). For details of
+the flat fielding correction, see :ref:`flat_field <flat_field_step>`.
```

Comment:
Need to change <flat_field_step> to <flatfield_step> (remove the first underscore). I gave you the wrong syntax in the first review 👎